### PR TITLE
Add missing routing tests

### PR DESCRIPTION
**Pull request by ggediminass**

### CODE REVIEW

File: test/qa/test_routing.py [hunk 1]
```diff
@@ -27,86 +29,145 @@ def setup_function(function):
...
+            assert SUBNET_2 in policy_routes
+            assert SUBNET_3 in policy_routes
 
+            network_interface = "nordtun" if tech == "openvpn" else "nordlynx"
+            assert network_interface in policy_routes
+
+            assert settings.get_is_routing_enabled()
+
+
+@pytest.mark.skip("LVPN-3273; LVPN-1574")
```
Comment (mariusSincovici) [inline test/qa/test_routing.py:67]: Could you add a comment to the tickets to enable the test when fixed?

File: test/qa/test_routing.py [hunk 2]
```diff
@@ -27,86 +29,145 @@ def setup_function(function):
...
+        sh.nordvpn.set.mesh.on()
+        assert "Routing is currently used by Meshnet. Disable it first." in sh.nordvpn.set.routing.off()
+        assert settings.get_is_routing_enabled()
+
+
+@pytest.mark.parametrize("tech,proto,obfuscated", lib.TECHNOLOGIES)
+def test_routing_already_enabled(tech, proto, obfuscated):
+    lib.set_technology_and_protocol(tech, proto, obfuscated)
+
+    assert "Routing is already set to 'enabled'." in sh.nordvpn.set.routing.on()
```
Comment (mariusSincovici) [inline test/qa/test_routing.py:162]: who is setting the routing here?

File: test/qa/test_routing.py (hunk 1 above)
Comment (ggediminass) [reply test/qa/test_routing.py:67]: Good point :smile:

File: test/qa/test_routing.py (hunk 2 above)
Comment (ggediminass) [reply test/qa/test_routing.py:162]: We are setting routing to enabled with `sh.nordvpn.set.routing.on()` in `assert "Routing is already set to 'enabled'." in sh.nordvpn.set.routing.on()`

File: test/qa/test_routing.py [hunk 3]
```diff
@@ -27,86 +29,152 @@ def setup_function(function):
...
-            assert "nordlynx" in policyRoutes
+def test_routing_enabled_connect(tech, proto, obfuscated):
+    lib.set_technology_and_protocol(tech, proto, obfuscated)
+
+    lib.add_subnet_to_allowlist(f"{SUBNET_1}/32")
+    lib.add_subnet_to_allowlist(f"{SUBNET_2}/32")
+    lib.add_subnet_to_allowlist(f"{SUBNET_3}/32")
+
+    with lib.Defer(sh.nordvpn.allowlist.remove.all):
+        with lib.Defer(sh.nordvpn.disconnect):
```
Comment (Savolro) [inline test/qa/test_routing.py:57]: Could we do multiple actions on a single defer instead of nesting them?

**Review by Savolro — CHANGES_REQUESTED** [review]

File: test/qa/test_routing.py (hunk 3 above)
Comment (ggediminass) [reply test/qa/test_routing.py:57]: I believe in that case we'd need to alter the lib.Defer itself, but it is possible

File: test/qa/lib/network.py [hunk 4]
```diff
@@ -20,6 +21,19 @@ def _is_internet_reachable(retry=5) -> bool:
     return False
 
 
+def _is_internet_not_reachable(retry=5) -> bool:
```
Comment (Savolro) [inline test/qa/lib/network.py:24]: Why not just use `not _is_internet_reachable()`?

File: test/qa/test_routing.py [hunk 5]
```diff
@@ -2,138 +2,232 @@
...
     network,
+    settings
 )
 import sh
 import lib
 import pytest
 import timeout_decorator
 
-def setup_module(module):
+# def setup_module(module):
```
Comment (Savolro) [inline test/qa/test_routing.py:17]: Please clean up commented out code

**Review by keliramu — APPROVED** [review]

**Review by bartoszWojciechO — APPROVED** [review]

**Pull request merged**